engine/src/main/java/org/terasology/particles/components/ParticleSystemComponent.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.particles.components;
+
+import org.lwjgl.util.vector.Vector2f;
+import org.terasology.entitySystem.Component;
+import org.terasology.entitySystem.Owns;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.particles.events.ParticleSystemUpdateEvent;
+import org.terasology.particles.rendering.ParticleRenderer;
+import org.terasology.rendering.assets.texture.Texture;
+
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Represents a particle system. A particle system must have emitters in order to produce particles.
+ */
+public class ParticleSystemComponent implements Component {
+
+    /**
+     * Reference to the entity this component is attached to
+     */
+    public EntityRef ownerEntity;
+
+    /**
+     * The renderer that should be used to render this system.
+     */
+    private Class<ParticleRenderer> renderer;
+
+    /**
+     * The lifetime of this system
+     */
+    public float maxLifeTime = Float.POSITIVE_INFINITY;
+
+    /**
+     * Destroy this particle system when its last emitter is destroyed TODO: Unimplemented
+     */
+    public boolean destroyWhenFinished = true;
```

Comment:
No worries. Let's remove it then: disposal functionality is very useful because particles can be involve a complex set of objects and relationships between them. Auto-disposal is not worth the effort in my opinion, or can be left for much later.